### PR TITLE
Fix Maxim drivers warnings

### PR DESCRIPTION
**Pull request by CiprianRegus**

### CODE REVIEW

File: drivers/platform/maxim/max32650/maxim_i2c.c [hunk 1]
```diff
@@ -40,6 +40,7 @@
 
 #include <stdint.h>
 #include <stdlib.h>
+#include <string.h>
```
Comment (amiclaus) [inline drivers/platform/maxim/max32650/maxim_i2c.c:43]: this change is unrelated to the commit. is this include required? same for all the other files changed in this commit.

Comment (buha) [reply drivers/platform/maxim/max32650/maxim_i2c.c:43]: i think it is, probably for strcpy() usage

Comment (buha) [reply drivers/platform/maxim/max32650/maxim_i2c.c:43]: or memcpy()

**Review by buha — APPROVED** [review]

File: drivers/platform/maxim/max32650/maxim_i2c.c (hunk 1 above)
Comment (CiprianRegus) [reply drivers/platform/maxim/max32650/maxim_i2c.c:43]: Indeed, it's for memcpy(), but I'll change the commit message, since it's totally unrelated.
